src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/AccountControllerTests.cs

```diff
@@ -0,0 +1,743 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Sources;
+using Castle.Core.Logging;
+using DevAdventCalendarCompetition.Controllers;
+using DevAdventCalendarCompetition.Models.Account;
+using DevAdventCalendarCompetition.Repository.Models;
+using DevAdventCalendarCompetition.Services.Interfaces;
+using FluentAssertions;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+using Moq;
+using Xunit;
+using static DevAdventCalendarCompetition.Resources.ViewsMessages;
+using SignInResult = Microsoft.AspNetCore.Identity.SignInResult;
+
+namespace DevAdventCalendarCompetition.Tests.UnitTests.Controllers
+{
+    public class AccountControllerTests
+    {
+        private readonly Mock<IAccountService> _accountServiceMock;
+        private readonly Mock<ILogger<AccountController>> _loggerMock;
+
+        public AccountControllerTests()
+        {
+            this._accountServiceMock = new Mock<IAccountService>();
+            this._loggerMock = new Mock<ILogger<AccountController>>();
+        }
+
+        [Fact]
+        public void Login_ReturnsLoginViewModel()
+        {
+            // Arrange
+            Uri returnUrl = null;
+            using var controller = new AccountController(this._accountServiceMock.Object, this._loggerMock.Object);
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { RequestServices = GetRequestService() }
+            };
+
+            // Act
+            var result = controller.Login(returnUrl);
```

Comment:
`Login` action in the controller is awaitable, so in the current solution we're not waiting for it to finish and the test will pass with the current assertion because it's basically true. To get the real `result` we should consider awaiting this action. Currently, we're mocking the `RequestServices` and by doing that we need to register all the services that will be used in that controller action. The problem with awaiting that action is that we'll get the exception `No service for type 'Microsoft.AspNetCore.Mvc.ViewFeatures.ITempDataDictionaryFactory' has been registered.`, so we need to register also that type.
So basically we should consider awaiting all the awaitable actions that we're testing, but if we're also mocking `RequestServices` we need to register all the services the particular test needs. That is the price of unit testing controllers :(